src/evolution.jl

```diff
@@ -20,37 +21,100 @@ include("constants.jl")
+                checkpoint_filename = joinpath(chkptdir, latest_checkpoint_file)
+            else
+                error("No checkpoint files found in directory: $chkptdir. Unable to recover.")
+            end
+
+            s, τ, N, B, L, N_sites, Δx, Δm², p, x, Δp, theta_nu, ψ, shape_name, energy_sign, cutoff, maxdim, t1, t2, t_initial, iteration = recover_checkpoint_hdf5(checkpoint_filename)
+            # Increment t_initial by τ to ensure it starts from the next expected value
+            t_initial += τ
+            s = siteinds(ψ)
+
+        elseif recover_type == "manual"
+            println("Manual recovery from iteration ", recover_iteration)
+            
+            # Create the checkpoint filename for the specified iteration
+            checkpoint_filename = joinpath(chkptdir, "checkpoint.chkpt.it" * lpad(recover_iteration, 6, "0") * ".h5")
+            
+            if isfile(checkpoint_filename)
+                println("Recovering from checkpoint: $checkpoint_filename")
+                # Recover data from the specified checkpoint
+                s, τ, N, B, L, N_sites, Δx, Δm², p, x, Δp, theta_nu, ψ, shape_name, energy_sign, cutoff, maxdim, t1, t2, t_initial, iteration = recover_checkpoint_hdf5(checkpoint_filename)
+                
+                # Increment t_initial by τ to ensure it starts from the next expected value
+                t_initial += τ
+                s = siteinds(ψ)
+            else
+                error("Checkpoint file not found for iteration $recover_iteration in directory: $chkptdir.")
+            end
+        end
+     
+    end    
+
+    # Create empty array s to... 
+    Sz_array = [] # to store sz values for all sites  
+    Sy_array = [] # to store sy values for all sites 
+    Sx_array = [] # to store sx values for all sites 
+    t_array = [] # to store t values (same for all sites)
+    prob_surv_array = []   # to store survival probability values for all sites 
     x_values = []  # to store x values for all sites 
     pₓ_values = [] # to store px vector values for all sites
-    ρₑₑ_array = Float64[] # to store ρₑₑ values
-    ρ_μμ_array = Float64[] # to store ρ_μμ values
-    ρₑμ_array = Float64[] # to store ρₑμ values
+    ρₑₑ_array = [] # to store ρₑₑ values for all sites 
+    ρ_μμ_array = [] # to store ρ_μμ values for all sites 
+    ρₑμ_array = [] # to store ρₑμ values for all sites 
     
-    ρₑμ_at_t1 = nothing  # Initialize a variable to store ρₑμ at t1
-    ρₑμ_at_t2 = nothing  # Initialize a variable to store ρₑμ at t2
-    Δt = t2 - t1 #time difference between growth rates 
-
-
-    # H = Hamiltonian_mpo(s, N, B, N_sites, Δx, Δm², p, x, Δp, shape_name,L, τ, energy_sign, periodic)
     # extract output of p_hat and p_mod for the p vector defined above for all sites. 
     p_mod, p̂ = momentum(p,N_sites) 
     p̂ₓ= [sub_array[1] for sub_array in p̂]
     
+    # Main evolution loop
+    is_first_iteration = true  # Flag to mark the first iteration of the loop
```

Comment:
Can this not be replaced with a line before the loop? You can just set:
t_recover = t_initial
and delete the if statement and the is_first_iteration variable